universal-application-tool-0.0.1/app/controllers/dev/FileUploadController.java

```diff
@@ -57,17 +59,30 @@ public Result create(Request request) {
     if (!isDevEnvironment()) {
       return notFound();
     }
-    Optional<String> bucket = request.queryString("bucket");
-    Optional<String> key = request.queryString("key");
     Optional<String> etag = request.queryString("etag");
-    if (!bucket.isPresent() || !key.isPresent()) {
-      return redirect(routes.FileUploadController.index().url());
+    String successMessage;
+    if (storageClient.getStorageServiceName() == StorageServiceName.AWS_S3) {
```

Comment:
suggestion (blocking): It looks like the `else` case here is implicitly Azure.

To clean this code up I recommend factoring out a [strategy pattern](https://en.wikipedia.org/wiki/Strategy_pattern) based on the file storage backend with a separate class for each. This will extend nicely when we add GCP.